doc/Architecture-overview/index.md

```diff
@@ -10,156 +10,157 @@ sort_idx: 15
+a [single node]({{root_url}}Administration-guide/System-installation/Single-node-installation/), or a [clustered deployment]({{root_url}}Administration-guide/System-installation/Cluster-setup/).
 
-![Logical concepts](attach/logical-concepts.png)
+An *application* in Kaa defines a set of data models, types of communication between between the endpoints and Kaa server, and processing rules.
+Kaa applications are not specific to the target platform, operating system, or the client software implementation.
+For example, two firmware implementations for a pressure sensor will differ between Arduino and STM32 platforms, yet will be considered the same application in Kaa as long as they report identically structured telemetry data.
+
+Kaa platform is multi-tenant.
+A single Kaa instance can support multiple independent business entities.
+Applications belong to tenants, while endpoints register within applications (see the picture below).
 
-*Kaa instance* (interchangeable with *Kaa deployment*) is a particular installation of the Kaa platform, either as
-a [single node]({{root_url}}Administration-guide/System-installation/Single-node-installation/),
-or a [clustered deployment]({{root_url}}Administration-guide/System-installation/Cluster-setup/).
+An [endpoint (EP)]({{root_url}}Glossary/#endpoint-ep) is an abstraction that represents a separately managed entity within a Kaa deployment.
+Practically speaking, an endpoint is a specific [Kaa client]({{root_url}}Glossary/#kaa-client) registered (or waiting to be registered) within a Kaa instance.
+Depending on the use case, different level physical entities can be considered endpoints.
+In an industrial setting, a single air quality sensor can represent an individual endpoint, while in a fleet tracking application, a truck (despite carrying
+on board multiple sensors that report data) may be a more appropriate entity to be declared as an endpoint.
 
-An *application* in Kaa defines the set of data models, the corresponding flows among the endpoints and Kaa server, and processing rules.
-Kaa applications are agnostic to the specific target platform, operating system, or the client software implementation.
-For example, two firmware implementations for a pressure sensor will differ between Arduino and STM32 platforms, yet will be considered the same application
-in Kaa, as long as they report identically structured telemetry data.
+![Logical concepts](attach/logical-concepts.png)
 
-The Kaa platform is *multi-tenant*, and may support multiple independent business entities out of a single instance.
-As illustrated in the preceding diagram, applications belong to tenants, while endpoints register within applications.
+To distinguish endpoints by different parameters and properties, rather than use one ID, Kaa uses [endpoint profiles]({{root_url}}Glossary/#endpoint-profile-client-side-server-side).
```

Comment:
"... by different parameters and properties,". Is it better to say "... by different properties," ?